source/use-cases/local-master-key.txt

```diff
@@ -1,23 +1,48 @@
 A. Create a Local Master Key
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Client-Side Field Level Encryption for MongoDB uses an encryption strategy called envelope encryption in which keys used to encrypt/decrypt data (called **data encryption keys**) are encrypted with another key (called the **master key**). 
+MongoDB `Client-Side Field Level Encryption (CSFLE) <https://docs.mongodb.com/manual/core/security-client-side-encryption/>`_ uses an encryption strategy called *envelope encryption* in which keys used to encrypt/decrypt data (called **data encryption keys**) are encrypted with another key (called the **master key**). For more information on the features of envelope encryption and key management concepts, see `AWS Key Management Service Concepts <https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#enveloping>`_.
 
-This strategy bolsters the security, minimizes encryption/decryption overhead, and allows the data encryption keys to be stored in the same replica set as the encrypted data. If a basic encryption strategy was used instead of envelope encryption, the computational and network load of encryption and decryption would need to be handled by the system that manages your keys. For more information on envelope encryption and key management concepts, read `this AWS article <https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#enveloping>`_.
+The master key, used by the MongoDB driver to create and encrypt data keys, should be stored remotely in a Key Management Service (KMS). The data encryption keys, generated and used by the MongoDB driver to encrypt and decrypt document fields, are stored in a key vault collection in the same database as the encrypted data.
 
-The master key is stored separately in a Key Management System (KMS) in order to prevent the data encryption keys from being used directly to access the sensitive data.
+In the following example, we forgo using a KMS for the master key and instead generate a local one for testing purposes. This local master key is a 96-byte, randomly generated string saved to a file.
```

Comment:
nit: passive voice -> active voice

"... randomly generated string that we save to a file."

Reply:
nit: typically it's best to keep "the following" as close as possible to the actual thing you're referring to. In this instance, it's not totally clear whether "the following example" is the code block or the rest of the page. If we put "the following" into the sentence just before the code block and end that sentence with a colon it may be clearer.

Maybe we could split this into two sentences:
- to make it easier to develop & test we'll initially use a local master key
- the following code randomly generates a 96-byte master key and saves it to a local file

We already have the content of the second bullet point in the tabbed section, so maybe just break out the first bullet point here

Reply:
Will update to convey the point about making it easier to develop and test by using a local master key.